bleak/_api.py

```diff
@@ -0,0 +1,214 @@
+        BleakGATTCharacteristicCoreBluetooth as _BleakGATTCharacteristicImplementation,
+    )  # noqa: F401
+    from bleak.backends.corebluetooth.descriptor import (
+        BleakGATTDescriptorCoreBluetooth as _BleakGATTDescriptorImplementation,
+    )  # noqa: F401
+elif platform.system() == "Windows":
+    # Requires Windows 10 Creators update at least, i.e. Window 10.0.16299
+    _vtup = platform.win32_ver()[1].split(".")
+    if int(_vtup[0]) != 10:
+        raise BleakError(
+            "Only Windows 10 is supported. Detected was {0}".format(
+                platform.win32_ver()
+            )
+        )
+
+    if (int(_vtup[1]) == 0) and (int(_vtup[2]) < 16299):
+        raise BleakError(
+            "Requires at least Windows 10 version 0.16299 (Fall Creators Update)."
+        )
+
+    from bleak.backends.winrt.scanner import (
+        BleakScannerWinRT as _BleakScannerImplementation,
+    )  # noqa: F401
+    from bleak.backends.winrt.device import (
+        BLEDeviceWinRT as _BLEDeviceImplementation,
+    )  # noqa: F401
+    from bleak.backends.winrt.client import (
+        BleakClientWinRT as _BleakClientImplementation,
+    )  # noqa: F401
+    from bleak.backends.winrt.service import (
+        BleakGATTServiceWinRT as _BleakGATTServiceImplementation,
+    )  # noqa: F401
+    from bleak.backends.winrt.characteristic import (
+        BleakGATTCharacteristicWinRT as _BleakGATTCharacteristicImplementation,
+    )  # noqa: F401
+    from bleak.backends.winrt.descriptor import (
+        BleakGATTDescriptorWinRT as _BleakGATTDescriptorImplementation,
+    )  # noqa: F401
+else:
+    raise BleakError(f"Unsupported platform: {platform.system()}")
+
+# Now let's tie together the abstract class and the backend implementation
+class BleakScanner(_BleakScannerImplementation):
+    """Interface for Bleak Bluetooth LE Scanners.
+
+    A BleakScanner can be used as an asynchronous context manager in which case it automatically
+    starts and stops scanning.
+
+    The actual implementation is dependent on the backend used, and the
+    constructor may have additional optional arguments.
+
+    :param detection_callback:
+            Optional function that will be called each time a device is
+            discovered or advertising data has changed.
+    :type detection_callback: Optional[Callable[[BLEDevice, AdvertisementData], Optional[Awaitable[NoneType]]]]
+    :param service_uuids:
+            Optional list of service UUIDs to filter on. Only advertisements
+            containing this advertising data will be received.
+    :type service_uuids: Optional[List[str]]
+    :param scanning_mode:
+            Set to "passive" to avoid the "active" scanning mode.
+    :type scanning_mode: Literal['active', 'passive']
+    """
+
+    # Gross hack to get the best docstring:
+    if len(__doc__) < len(_BleakScannerImplementation.__doc__):
+        __doc__ = _BleakScannerImplementation.__doc__
```

Comment:
Yes, this doesn't seem like something we want.

Reply:
Fully agreed. But we somehow need to distinguish between two cases, and this was the quick-and-dirty way I came up with:

- If the base class is incomplete (i.e. we're building for RTD) we should provide the docstring for the constructor,
- if the base class is complete (i.e. we're at runtime, and we need to provide help for introspection tools and `help()` it's better to provide the base class docstring because it's better.

Suggestions?

Reply:
Fixed, by simply adding a mock base class for RTD. Also has the advantage that we have a constructor signature now.